shared-model-classes/src/main/java/com/exasol/projectkeeper/shared/config/workflow/WorkflowStep.java

```diff
@@ -0,0 +1,78 @@
+package com.exasol.projectkeeper.shared.config.workflow;
+
+import java.util.*;
+
+/**
+ * A build step in a GitHub workflow.
+ */
+public final class WorkflowStep {
+
+    private final Map<String, Object> rawStep;
+
+    private WorkflowStep(final Map<String, Object> rawStep) {
```

Comment:
Is there a specific reason for the constructor being private and offering a static method for instantiation?

Reply:
We might want to add some validation in the future.